src/pages/MyCoursePage.tsx

```diff
@@ -0,0 +1,327 @@
+                        description: 'Test #3',
+                        weight: 40,
+                        type: 'Schriftliche Prüfung',
+                    },
+                ],
+                teachers: [
+                    {
+                        id: 2,
+                        name: 'John Doe',
+                    },
+                ],
+            },
+            {
+                id: 3,
+                description: 'Software-Architektur',
+                exams: [
+                    {
+                        id: 7,
+                        description: 'Test #1',
+                        weight: 50,
+                        type: 'Schriftliche Prüfung',
+                    },
+                    {
+                        id: 8,
+                        description: 'Test #2',
+                        weight: 30,
+                        type: 'Mündliche Prüfung',
+                    },
+                    {
+                        id: 9,
+                        description: 'Test #3',
+                        weight: 20,
+                        type: 'Mündliche Prüfung',
+                    },
+                ],
+                teachers: [
+                    {
+                        id: 2,
+                        name: 'John Doe',
+                    },
+                ],
+            },
+        ],
+    },
+];
+
+export default function MyCoursePage() {
+    const dropdownItems = customCourses.map((customCourse, index) => ({
+        label: customCourse.startDate,
+        value: index,
+    }));
+
+    type MenuItem = Required<MenuProps>['items'][number];
+
+    const getItem = React.useCallback(
+        (
+            label: React.ReactNode,
+            key: React.Key,
+            icon?: React.ReactNode,
+            children?: MenuItem[],
+            type?: 'group'
+        ): MenuItem => {
+            return {
+                key,
+                icon,
+                children,
+                label,
+                type,
+            } as MenuItem;
+        },
+        []
+    );
+
+    const getMenuItems = React.useCallback(
+        (key: number) => {
+            return customCourses[key].course.map((course, index) =>
+                getItem(course.description, index.toString(), <BookFilled />)
+            );
+        },
+        [getItem]
+    );
+
+    const intl = useIntl();
+    const [menuItems, setMenuItems] = useState<MenuItem[]>(getMenuItems(0));
+    const [current, setCurrent] = useState<MenuItemType>(menuItems[0] as MenuItemType);
+
+    const handleOnChange = React.useCallback(
+        (e: number) => {
+            setMenuItems(getMenuItems(e));
+        },
+        [getMenuItems]
+    );
+
+    const menuOnClick = React.useCallback(
+        (e: MenuInfo) => {
+            setCurrent(menuItems[parseInt(e.key)] as MenuItemType);
+        },
+        [menuItems]
+    );
+
+    // Automatically select the first menu item when the selected year changes
```

Comment:
your comment says you change the selection when the year changes but you listen on menuItems and not on years, is that correct?

Reply:
Correct, the menuItems are changing whenever another year is selected (see handleOnChange), so I think I can listen on menuItems. If it's better to listen to the dropdownItems/years I'll definitely change it.

Reply:
Ah okay now I see. Well I would make the following changes:
Move the getItem and getMenuItems outside of the comment, since they are not depending on state. Also then you do not need the usecallback. Then get rid of the useEffect for setting the first menu item when year changes and instead setCurrent also in the handeOnchange to the first element.